File: seo/04w/04w_boj_15649.py
import sys

# 중복 제거

def rec(depth, n, m):
    if depth == m:
        print(*sequence)
        return
    for i in range(1, n+1, 1):
        sequence[depth] = i
        rec(depth+1, n, m)


N, M = map(int, sys.stdin.readline().split())

sequence = [0] * M
rec(0, N, M)

File: mo/04w/04w_boj_15649.c
#include <stdio.h>

int v[10];
int arr[10];

void rec(int depth, int n, int m) {
    if (depth == m) {
        for (int i = 0; i < m; ++i) {
            printf("%d ", arr[i]);
        }
        printf("\n");
        return ;
    }
    for (int i = 1; i <= n; ++i) {
        if (!v[i]) {
            v[i] = 1;
            arr[depth] = i;
            rec(depth + 1, n, m);
            v[i] = 0;
        }
    }
}

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    rec(0, n, m);
    return 0;
}
